socialhome/streams/app/components/ProfileReactionButtons.vue

```diff
@@ -72,25 +81,33 @@ export default {
                 unfollow: gettext("Unfollow"),
             }
         },
+        urls() {
+            return {
+                followUrl: Urls["api:profile-add-follower"]({pk: this.currentBrowsingProfileId}),
+                unfollowUrl: Urls["api:profile-remove-follower"]({pk: this.currentBrowsingProfileId}),
+            }
+        },
     },
     methods: {
         follow() {
-            this.$http.post(
-                Urls["api:profile-add-follower"]({pk: this.currentBrowsingProfileId}),
-                {guid: this.profile.guid})
-                .then(() => {
-                    this.following = true
-                })
-                .catch(err => console.error(err) /* TODO: Proper error handling */)
+            if (!this.isUserAuthenticated) {
```

Comment:
I think it's OK to add these, but it's unnecessary effort. We don't render the buttons if the user is not logged in. So, unless someone makes a mess of the code the user will never see this notification. And even if someone does, the backend will still validate and raise an error. Better have some kind of central error notify for these kinds of things rather than raising errors that will never be raised as the code cannot be reached.

Let's leave them in now as they are here, but for the future, to save your time ;)

Reply:
> I think it's OK to add these, but it's unnecessary effort.

If shit can be done with your app, shit *will* be done with you app ;)

Reply:
Not on my watch... 😁

Reply:
![dd505ffbfd7b5dc7f7e0fd75fbc5e8cb17c8f722ce54ab1ba2386fbb3d31a6f8](https://user-images.githubusercontent.com/22097904/35474986-e475ab1c-0396-11e8-9b68-a96070d461fd.jpg)